webpack/assets/javascripts/react_app/components/Layout/components/TaxonomySwitcher/TaxonomySwitcher.js

```diff
@@ -0,0 +1,80 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { ToolbarItem, Spinner } from '@patternfly/react-core';
+import { noop } from '../../../../common/helpers';
+
+import { locationPropType, organizationPropType } from '../../LayoutHelper';
+import {
+  ANY_ORGANIZATION_TEXT,
+  ANY_LOCATION_TEXT,
+} from '../../LayoutConstants';
+import TaxonomyDropdown from './TaxonomyDropdown';
+
+const TaxonomySwitcher = ({
+  currentOrganization,
+  currentLocation,
+  organizations,
+  locations,
+  taxonomiesBool,
+  isLoading,
+  onLocationClick,
+  onOrgClick,
+}) => (
+  <React.Fragment>
+    {taxonomiesBool.organizations && (
+      <ToolbarItem>
+        <TaxonomyDropdown
+          taxonomyType="Organization"
+          id="organization-dropdown"
+          currentTaxonomy={currentOrganization}
+          taxonomies={organizations}
+          changeTaxonomy={onOrgClick}
+          anyTaxonomyText={ANY_ORGANIZATION_TEXT}
+          manageTaxonomyText="Manage Organizations"
+          anyTaxonomyURL="/organizations/clear"
+          manageTaxonomyURL="/organizations"
+        />
+      </ToolbarItem>
+    )}
+    {taxonomiesBool.locations && (
+      <ToolbarItem>
+        <TaxonomyDropdown
+          taxonomyType="Location"
+          id="location-dropdown"
+          currentTaxonomy={currentLocation}
+          taxonomies={locations}
+          changeTaxonomy={onLocationClick}
+          anyTaxonomyText={ANY_LOCATION_TEXT}
+          manageTaxonomyText="Manage Locations"
+          anyTaxonomyURL="/locations/clear"
+          manageTaxonomyURL="/locations"
+        />
+      </ToolbarItem>
+    )}
+    {isLoading && <Spinner size="md" />}
```

Comment:
is it `isLoading` for the application loading spinner?
if it's loading of the dropdown's data can you use a `skeleton` ?

Reply:
The loading is for the page loading and not the dropdowns data. For example, the spinner is used when creating a new host and selecting an architecture